src/MapboxStyleParser.ts

```diff
@@ -0,0 +1,1087 @@
+        if (typeof label === 'string') {
+            return label;
+        }
+        if (label[0] !== 'format') {
+            throw new Error(`Cannot parse mapbox style. Unsupported text format.`);
+        }
+        let gsLabel = '';
+        // ignore all even indexes since we cannot handle them 
+        for (let i = 1; i < label.length; i = i + 2) {
+            if (typeof label[i] === 'string') {
+                gsLabel += label[i];
+            } else {
+                if (label[i][0] !== 'get') {
+                    throw new Error(`Cannot parse mapbox style. Unsupported lookup type.`);
+                }
+                gsLabel += '{{' + label[i][1] + '}}';
+            }
+        }
+        return gsLabel;
+    }
+
+    /**
+     * Creates a GeoStylerStyle-MarkSymbolizer from a Mapbox Style Layer
+     *
+     * @param {any} layer A Mapbox Style Layer
+     * @return {MarkSymbolizer} A GeoStylerStyle-MarkSymbolizer
+     */
+    getMarkSymbolizerFromMapboxLayer(paint: any): MarkSymbolizer {
+        // TODO parse MarkSymbolizer
+        return {
+            kind: 'Mark',
+            wellKnownName: 'Circle'
+        };
+    }
+
+    /**
+     * Creates a GeoStylerStyle-IconSymbolizer from a Mapbox Style Layer
+     *
+     * @param {any} layer A Mapbox Style Layer
+     * @return {IconSymbolizer} A GeoStylerStyle-IconSymbolizer
+     */
+    getIconSymbolizerFromMapboxLayer(paint: any): IconSymbolizer {
+        // TODO parse IconSymbolizer
+        return {
+            kind: 'Icon'
+        };
+    }
+
+    /**
+     * Creates a GeoStylerStyle-TextSymbolizer from a Mapbox Style Layer
+     *
+     * @param {any} layer A Mapbox Style Layer
+     * @return {TextSymbolizer} A GeoStylerStyle-TextSymbolizer
+     */
+    getTextSymbolizerFromMapboxLayer(paint: any): TextSymbolizer {
+        return {
+            kind: 'Text',
+            spacing: paint['symbol-spacing'],
+            avoidEdges: paint['symbol-avoid-edges'],
+            pitchAlignment: paint['text-pitch-alignment'],
+            rotationAlignment: paint['text-rotation-alignment'],
+            label: this.getLabelFromTextField(paint['text-field']),
+            font: paint['text-font'],
+            size: paint['text-size'],
+            maxWidth: paint['text-max-width'],
+            lineHeight: paint['text-line-height'],
+            letterSpacing: paint['text-letter-spacing'],
+            justify: paint['text-justify'],
+            anchor: paint['text-anchor'],
+            maxAngle: paint['text-max-angle'],
+            rotate: paint['text-rotate'],
+            padding: paint['text-padding'],
+            keepUpright: paint['text-keep-upright'],
+            transform: paint['text-transform'],
+            offset: paint['text-offset'],
+            allowOverlap: paint['text-allow-overlap'],
+            ignorePlacement: paint['text-ignore-placement'],
+            optional: paint['text-optional'],
+            visibility: paint.visibility,
```

Comment:
Very minor: Please use bracket notation here as well.

Reply:
Solved in https://github.com/terrestris/geostyler-mapbox-parser/pull/1/commits/b8912d60e00672bff2a78e67b869744c5cb9f8b6